stream/management/commands/pittgoogle.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+# -*- coding: UTF-8 -*-
+"""Listen to a Pitt-Google Pub/Sub stream and save alerts to the database."""
+from importlib import import_module
+import logging
+
+from django.conf import settings
+from django.core.management.base import BaseCommand
+from django.db import transaction
+
+from stream.models import Alert, Topic
+
+from pittgoogle import Consumer
+
+
+logger = logging.getLogger(__name__)
+
+PITTGOOGLE_CONSUMER_CONFIGURATION = settings.PITTGOOGLE_CONSUMER_CONFIGURATION
+PITTGOOGLE_PARSERS = settings.PITTGOOGLE_PARSERS
+
+
+def get_parser_classes(topic):
+    """."""
+    parser_classes = []
+
+    try:
+        parsers = PITTGOOGLE_PARSERS[topic]
+    except KeyError:
+        logger.log(msg=f'PITTGOOGLE_PARSERS not found for topic: {topic}.', level=logging.WARNING)
+        return parser_classes
+
+    for parser in parsers:
+        mod_name, class_name = parser.rsplit('.', 1)
+        try:
+            mod = import_module(mod_name)
+            clazz = getattr(mod, class_name)
+        except (ImportError, AttributeError):
+            raise ImportError(f'Unable to import parser {parser}')
+        parser_classes.append(clazz)
+
+    return parser_classes
+
+
+class Command(BaseCommand):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.consumer = Consumer(PITTGOOGLE_CONSUMER_CONFIGURATION['subscription_name'])
+
+    def handle(self, *args, **options):
+        """Pull and process alerts from a Pitt-Google Pub/Sub stream.
+
+        -   Execute a "streaming pull" in a background thread.
+            Asyncronously, pull alerts and process them through the callback,
+            `parse_and_save`.
+
+        -   Stop when a stoping condition is reached.
+
+        -   Block until the processing is complete and the connection has been closed.
+        """
+        self.consumer.stream_alerts(
+            callback=self.parse_and_save,
+            stop_conditions=PITTGOOGLE_CONSUMER_CONFIGURATION['stop_conditions'],
+        )
```

Comment:
If you would really prefer to iterate over the alerts, instead of running your logic in the background thread, the above command would simply change to:

```python
alert_list = self.consumer.stream_alerts(stop_conditions=stop_conditions)
for alert in alert_list:
    self.parse_and_save(alert, self.consumer.topic_name)
```

I see 2 disadvantages to iterating over the alerts in this way:

1. Alerts will be "acknowledged" to Pub/Sub (and therefore leave the subscription) _before_ your logic is run. If there is some error in the interim, you could lose a lot of alerts.
2. All alerts will be held in memory at once, since they will be collected in the background and returned to you after collection is complete.

I _could_ write some more complicated logic to avoid these two disadvantages and still let you process in the main thread by iterating over alerts, but I don't see a reason to make it that complicated. (But maybe you do see a reason to do it this way?)